mllib/src/main/scala/org/apache/spark/ml/param/sharedParams.scala

```diff
@@ -17,6 +17,10 @@
 
 package org.apache.spark.ml.param
 
+/* NOTE TO DEVELOPERS:
+ * If you add these parameter traits into your algorithm, you need to add a setter method as well.
```

Comment:
Maybe we should update this comment and explain *why* the setter must be added? Code will still compile, right?